File: iOS/Athlynk/AthlynkCoach/Features/CoachMoreView.swift
//
//  CoachMoreView.swift
//  "Altro" hub — entry point to the secondary coach areas: athletes, chat,
//  agenda, subscriptions, resource library, analytics, notifications and the
//  coach profile. (Allenamento / Nutrizione / Check are now primary tabs.)
//

import SwiftUI

enum CoachRoute: Hashable, Identifiable {
    case clients, chat, agenda, subscriptions, resources, analytics, notifications, profile
    var id: Self { self }
}

struct CoachMoreView: View {
    @EnvironmentObject private var app: AppState
    @Binding var path: NavigationPath

    private struct Item: Identifiable {
        let route: CoachRoute
        var id: CoachRoute { route }
        let icon: String
        let title: String
        let subtitle: String
        let accent: Color
    }

    private let items: [Item] = [
        .init(route: .clients, icon: "person.2.fill", title: "Atleti",
              subtitle: "I tuoi clienti", accent: Palette.cyan),
        .init(route: .chat, icon: "bubble.left.fill", title: "Chat",
              subtitle: "Messaggi con gli atleti", accent: Palette.violet),
        .init(route: .agenda, icon: "calendar", title: "Agenda",
              subtitle: "Appuntamenti", accent: Palette.amber),
        .init(route: .subscriptions, icon: "creditcard.fill", title: "Abbonamenti",
              subtitle: "Piani e ricavi", accent: Palette.bronze),
        .init(route: .resources, icon: "books.vertical.fill", title: "Libreria Risorse",
              subtitle: "Modelli riutilizzabili", accent: Palette.violet),
        .init(route: .analytics, icon: "chart.line.uptrend.xyaxis", title: "Analisi Progressi",
              subtitle: "Metriche dello studio", accent: Palette.cyan),
        .init(route: .notifications, icon: "bell.fill", title: "Notifiche",
              subtitle: "Centro notifiche", accent: Palette.phase),
        .init(route: .profile, icon: "person.crop.square.fill", title: "Profilo & Impostazioni",
              subtitle: "Il tuo profilo pubblico", accent: Palette.bronze),
    ]

    @State private var appear = false

    var body: some View {
        NavigationStack(path: $path) {
            ScreenScroll {
                ScreenHeader(eyebrow: "Studio", title: "Altro",
                             subtitle: "Strumenti e gestione", accent: Palette.phase)
                    .revealUp(appear, index: 0)

                LazyVGrid(columns: [GridItem(.flexible(), spacing: 14), GridItem(.flexible(), spacing: 14)],
                          spacing: 14) {
                    ForEach(Array(items.enumerated()), id: \.element.id) { i, item in
                        Button { Haptics.tap(); path.append(item.route) } label: { card(item) }
                            .buttonStyle(PressableButtonStyle())
                            .revealUp(appear, index: i / 2 + 1)
                    }
                }
            }
            .onAppear { appear = true }
            .navigationDestination(for: CoachRoute.self) { route in
                switch route {
                case .clients:       CoachClientsView()
                case .chat:          CoachMessagesView()
                case .agenda:        CoachAgendaView()
                case .subscriptions: CoachSubscriptionsView()
                case .resources:     CoachResourcesView()
                case .analytics:     CoachAnalyticsView()
                case .notifications: CoachNotificationsView()
                case .profile:       CoachProfileView()
                }
            }
        }
    }

    private func card(_ item: Item) -> some View {
        HubTile(icon: item.icon, title: item.title, subtitle: item.subtitle, accent: item.accent)
    }
}

File: android/lib/features/coach/more/coach_more_view.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../../core/providers.dart';
import '../../../design/components/avatar.dart';
import '../../../design/components/panel.dart';
import '../../../design/components/scaffold.dart';
import '../../../design/theme.dart';
import '../../athlete/notifications/notifications_view.dart';
import '../agenda/coach_agenda_view.dart';
import '../analytics/coach_analytics_view.dart';
import '../clients/coach_clients_view.dart';
import '../messages/coach_messages_view.dart';
import '../profile/coach_profile_view.dart';
import '../resources/coach_resources_view.dart';
import '../subscriptions/coach_subscriptions_view.dart';

/// Coach "Altro" hub — port of iOS `CoachMoreView`: identity card + 8-tile
/// grid over the secondary destinations.
class CoachMoreView extends ConsumerWidget {
  const CoachMoreView({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final session = ref.watch(sessionControllerProvider);
    final user = session.user;

    void push(Widget screen) => Navigator.of(context)
        .push(MaterialPageRoute<void>(builder: (_) => screen));

    final tiles = <({
      String title,
      String subtitle,
      IconData icon,
      Color accent,
      VoidCallback onTap
    })>[
      (
        title: 'Atleti',
        subtitle: 'Il tuo roster',
        icon: Icons.groups_rounded,
        accent: Palette.bronze,
        onTap: () => push(const CoachClientsView()),
      ),
      (
        title: 'Chat',
        subtitle: 'Parla con i tuoi atleti',
        icon: Icons.forum_rounded,
        accent: Palette.violet,
        onTap: () => push(const CoachMessagesView()),
      ),
      (
        title: 'Agenda',
        subtitle: 'Appuntamenti e visite',
        icon: Icons.calendar_month_rounded,
        accent: Palette.cyan,
        onTap: () => push(const CoachAgendaView()),
      ),
      (
        title: 'Abbonamenti',
        subtitle: 'Piani, incassi e Stripe',
        icon: Icons.workspace_premium_rounded,
        accent: Palette.amber,
        onTap: () => push(const CoachSubscriptionsView()),
      ),
      (
        title: 'Libreria risorse',
        subtitle: 'Schede, piani, modelli',
        icon: Icons.folder_copy_rounded,
        accent: Palette.lime,
        onTap: () => push(const CoachResourcesView()),
      ),
      (
        title: 'Analisi',
        subtitle: 'KPI e rischio abbandono',
        icon: Icons.query_stats_rounded,
        accent: Palette.bronze,
        onTap: () => push(const CoachAnalyticsView()),
      ),
      (
        title: 'Notifiche',
        subtitle: 'Centro notifiche',
        icon: Icons.notifications_rounded,
        accent: Palette.amber,
        onTap: () => push(const NotificationsView()),
      ),
      (
        title: 'Profilo & Impostazioni',
        subtitle: 'I tuoi dati e preferenze',
        icon: Icons.person_rounded,
        accent: Palette.phase,
        onTap: () => push(const CoachProfileView()),
      ),
    ];

    return ScreenScroll(
      spacing: Space.element,
      children: [
        const ScreenHeader(eyebrow: 'La tua cabina di regia', title: 'Altro'),
        VoltPanel(
          child: Row(
            children: [
              AvatarView(
                url: session.avatarUrl,
                name: user?.displayName ?? 'Coach',
                size: 52,
              ),
              const SizedBox(width: 14),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(user?.displayName ?? 'Coach',
                        style: Typo.display(18)),
                    Text(user?.email ?? '',
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                        style: Typo.body(
                            12.5, FontWeight.w400, Palette.textMid)),
                    const SizedBox(height: 4),
                    StatusBadge('Coach', color: Palette.bronze),
                  ],
                ),
              ),
            ],
          ),
        ),
        const Eyebrow('Gestione'),
        GridView.count(
          crossAxisCount: 2,
          shrinkWrap: true,
          physics: const NeverScrollableScrollPhysics(),
          mainAxisSpacing: 12,
          crossAxisSpacing: 12,
          childAspectRatio: 1.18,
          children: [
            for (final (i, t) in tiles.indexed)
              HubTile(
                title: t.title,
                subtitle: t.subtitle,
                icon: t.icon,
                accent: t.accent,
                index: i,
                onTap: t.onTap,
              ),
          ],
        ),
      ],
    );
  }
}
